Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -336,21 +336,42 @@ def pre_pr_checks(cls, repository):
             num_checks += 1
             name = key.split('.')[-1]
             log.info('    Running {}...'.format(name))
-            command_line = path.split(' ')
-            if command_line[0] == 'python3' and os.name == 'nt':
-                command_line[0] = sys.executable
-            command = run(command_line, cwd=repository.root_path)
-            if command.returncode:
-                if Terminal.choose(
+            for attempt in range(attempts):
+                command_line = path.split(' ')
+                if command_line[0] == 'python3' and os.name == 'nt':
+                    command_line[0] = sys.executable
+                command = run(command_line, cwd=repository.root_path)
+                if command.returncode == 0:
+                    log.info('    Ran {}!'.format(name))
+                    break
+                options = ['Continue', 'Re-Run', 'No']
+                response = Terminal.choose(
                     '{} failed, continue uploading pull request?'.format(name),
+                    options=(options[0], options[2]) if attempt + 1 == attempts else options,
                     default='No',
-                ) == 'No':
+                )
+                if response == 'No':
                     sys.stderr.write('Pre-PR check {} failed\n'.format(name))
                     return False
-                else:
+                if response == 'Continue':
                     log.info('    {} failed, continuing PR upload anyway'.format(name))
-            else:
-                log.info('    Ran {}!'.format(name))
+                    break
+
+                modified = [] if add_edits is False else repository.modified()
+                if add_edits:
+                    modified = list(set(modified).union(set(repository.modified(staged=False))))
+                for file in set(modified) - set(repository.modified(staged=True)):
+                    log.info('    Adding {}...'.format(file))
+                    if run([repository.executable(), 'add', file], cwd=repository.root_path).returncode:
+                        sys.stderr.write("Failed to add '{}'\n".format(file))
+                        return 1
```

Comment:
What's the return value convention in this function? `True / False` for `pass / fail`, how about 1? Do you mean `True` here?

Reply:
We're checking `returncode`, which is the return code of the process, so "pass" is `0` and "fail" is `1`.

Reply:
Good catch, this should be `return False`.